pkg/types/aws/platform.go

```diff
@@ -101,6 +101,12 @@ type Platform struct {
 	//
 	// +optional
 	LBType configv1.AWSLBType `json:"lbType,omitempty"`
+
+	// IgnoreS3Deletion is an optional field that can be used to make the S3 deletion optional
+	// during bootstrap destroy.
+	// See: https://issues.redhat.com/browse/CORS-2661
+	// +optional
+	IgnoreS3Deletion bool `json:"ignoreS3Deletion,omitempty"`
```

Comment:
What about using a suggested name for the bootstrap bucket? Are we using the bucket for other things than generating pre-signed URLs for bootstrap user data?

Reply:
Added a commit that allows users to change name

Reply:
@patrickdillon The only benefit I can see providing a bucket name in IC is to reuse the existing one, then when provided the installer could create an object path `infraID/bootstrap.ign` or even `infraID-bootstrap.ign` and generate the presigned URL for this object (maybe failing when already exists), so the customer can keep a single bucket with their N deployments, instead of N buckets - when the flag `PreserveBootstrapIgnition` is set.

My other suggestion is the var name, instead of using "S3", use "Bucket" - considering what is deleted is the instance (Bucket), and not the service (S3): `IgnoreBucketDeletion`